packages/vite/src/node/build.ts

```diff
@@ -435,13 +443,17 @@ export async function resolveBuildPlugins(config: ResolvedConfig): Promise<{
   }
 }
 
+let rollupWatcher: RollupWatcher
```

Comment:
I think we need to use a weakmap of the builds `resolvedConfig` to `RollupWatcher`, so that multiple build watches don't conflict with this single variable.